src/authentication/authentication-hapi-plugin.ts

```diff
@@ -0,0 +1,154 @@
+
+import * as Boom from 'boom';
+import * as auth from 'hapi-auth-jwt2';
+import { inject, injectable } from 'inversify';
+import * as Joi from 'joi';
+import * as qs from 'querystring';
+
+import * as Hapi from '../server/hapi';
+import { HapiPlugin } from '../server/hapi-register';
+import { Group, User } from '../shared/gitlab';
+import { GitlabClient } from '../shared/gitlab-client';
+import { Logger, loggerInjectSymbol } from '../shared/logger';
+import {default as AuthenticationModule} from './authentication-module';
+import { AccessToken, authServerBaseUrlInjectSymbol, jwtOptionsInjectSymbol } from './types';
+
+interface UserId {
+  id: string;
+  idp: string;
+}
+
+@injectable()
+class AuthenticationHapiPlugin extends HapiPlugin {
+
+  public static injectSymbol = Symbol('authentication-hapi-plugin');
+  private readonly authenticationModule: AuthenticationModule;
+  private readonly authServerBaseUrl: string;
+  private readonly logger: Logger;
+  private readonly gitlab: GitlabClient;
+  private readonly options: auth.JWTStrategyOptions;
+  private teamTokenClaimKey: 'https://team_token' = 'https://team_token';
```

Comment:
By the way, had you tried using a claim key that didn't start with `https://` in Auth0? It looks really, really weird in its current form.